src/test/java/tests/conformance/AbstractConformanceTest.java

```diff
@@ -104,15 +97,20 @@ private ConformanceTestReport runTests() throws IOException {
           .filter(path -> path.toFile().isDirectory())
           .flatMap(this::javaFileGroups)
           .flatMap(this::analyzeFiles)
-          .collect(collectingAndThen(toImmutableSet(), ConformanceTestReport::new));
+          .reduce(ConformanceTestReport.EMPTY, ConformanceTestReport::combine);
     }
   }
 
-  private Stream<ConformanceTestResult> analyzeFiles(List<Path> files) {
-    ImmutableMap<Path, ListMultimap<Long, ReportedFact>> reportedFactsByFile =
+  private Stream<ConformanceTestReport> analyzeFiles(List<Path> files) {
+    ImmutableListMultimap<Path, ReportedFact> reportedFactsByFile =
         Streams.stream(analyze(ImmutableList.copyOf(files)))
-            .collect(ReportedFact.BY_FILE_AND_LINE_NUMBER);
-    return files.stream().flatMap(file -> testResultsForFile(file, reportedFactsByFile).stream());
+            .collect(toImmutableListMultimap(ReportedFact::getFile, identity()));
```

Comment:
```suggestion
            .collect(toImmutableListMultimap(ReportedFact::getFile, rf -> rf));
```

mostly just for consistency with another file in the PR :)

Reply:
I got rid of all the `identity()`s, and used `Multimaps.index()` to simplify further.